### PR TITLE
Flask monitor library

### PR DESCRIPTION
**Pull request by gutorc92**
Signed-off-by: Gustavo Coelho <gutorc@hotmail.com>

### CODE REVIEW

File: flask_monitor/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+regex==2020.1.8
+requests==2.22.0
+safety==1.8.5
+six==1.14.0
+toml==0.10.0
+typed-ast==1.4.1
+typing-extensions==3.7.4.1
+urllib3==1.25.8
+wcwidth==0.1.8
+Werkzeug==0.16.1
```
Comment (abilioesteves) [inline flask_monitor/requirements.txt]: @gutorc92, do we really need all these requirements?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  - "3.6"      # current default Python on Travis CI
+  - "3.7"
+  - "3.8"
+  - "3.8-dev"  # 3.8 development branch
+# command to install dependencies
+install:
+  - pip install -r requirements.txt
+# command to run tests
+script:
+  - pytest
```
Comment (abilioesteves) [inline .travis.yml]: We'll configure our CI/CD pipeline with github action. Please remove this file.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,122 @@
...
+# Metrics
+
+The only exposed metrics (for now) are the following:
+
+```
+request_seconds_bucket{type,status, method, addr, version, isError, le}
+request_seconds_count{type, status, method, addr, version, isError}
+request_seconds_sum{type, status, method, addr, version, isError}
+response_size_bytes{type, status, method, addr, version, isError}
+dependency_up{name}
```
Comment (abilioesteves) [inline README.md]: @gutorc92, this is an old Big Brother protocol version. Please update your flask-monitor to adhere to Big Brother Metric Protocol v0.2.0.

**Comment by abilioesteves** [comment]
@gutorc92, why do you have 6 different examples? 

1. `flask-monitor/example.py`;
2. `example/app.py`;
3. `example/app_200_error.py`;
4. `example/bd_dependency.py`;
5. `example/dependency_metric_example.py;
6. `example/simple_example.py`;

I think a single, well documented, example that explores all important features - custom buckets config, custom error callback and dependency checking - should suffice.

Also, I think your lib API is lacking the possibility to configure weather or not the user wants to expose default Python metrics.

File: .travis.yml (hunk 2 above)
Comment (gutorc92) [reply .travis.yml]: I already changed the project to use github actions.

File: flask_monitor/requirements.txt (hunk 1 above)
Comment (gutorc92) [reply flask_monitor/requirements.txt]: Requirements file was generated by pipenv. Pipenv write down all packages used on the project.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,147 @@
...
+
+For a specific request:
+
+1. `type` tells which request protocol was used (e.g. `grpc`, `http`, `<your custom protocol>`);
+2. `status` registers the response status code; 
+3. `isError` let you know if the request's response status is considered an error;
+4. `method` registers the request method (e.g. `GET` for http get requests);
+5. `addr` registers the requested endpoint address;
+6. and `version` tells which version of your service has handled the request;
+
```
Comment (abilioesteves) [inline README.md]: @gutorc92, this seems redundant with the text from the previous section. Please, choose only one.

File: flask_monitor/example.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+if __name__ == "__main__":
+    run_simple(
+        "localhost",
+        5000,
+        create_dispatcher(),
+        use_reloader=True,
+        use_debugger=True,
+        use_evalex=True,
+    )
```
Comment (abilioesteves) [inline flask_monitor/example.py]: @gutorc92, could please explain why we still have two different examples? What about the example from the `example` folder? How is it different from this one?

File: example/simple_example.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+def hello_world():
+    return 'Hello, World!'
+
+@app.route('/database')
+def bd_running():
+    return 'I am a database working.'
+
+if __name__ == "__main__":
+    run_simple(hostname="localhost", port=5000, application=dispatcher)
+    # app.run()
```
Comment (abilioesteves) [inline example/simple_example.py]: @gutorc92, I need you to add comments to better explain each step. Also, please indicate what is optional and what is required.

File: flask_monitor/example.py (hunk 5 above)
Comment (gutorc92) [reply flask_monitor/example.py]: I forget to change the file name, it should be test_example. It is a file for tests, I also change this file location.

File: example/simple_example.py [hunk 7]
```diff
@@ -27,16 +38,23 @@ def check_db():
         traceback.print_stack()
     return 0
 
+# watch dependency
+# first parameter is the dependency's name. It's a mandatory parameter.
+# second parameter is the health check function. It's a mandatory parameter.
+# time_execution is a optional parameter
```
Comment (abilioesteves) [inline example/simple_example.py]: @gutorc92, what is the meaning of the `time_execution` param?

**Pull request merged**